gui/src/app/Compilation/Context/CompileContextProvider.tsx

```diff
@@ -8,13 +7,42 @@ import {
   useEffect,
   useState,
 } from "react";
-import { CompileContext, CompileStatus } from "./CompileContext";
-import { publicCompilationServerUrl } from "@SpStanc/CompilationServerConnectionControl";
+import { CompileContext, CompileStatus } from "@SpCompilation/CompileContext";
+import { publicCompilationServerUrl } from "@SpCompilation/Constants";
+import compileStanProgram from "./compileStanProgram";
 
 type CompileContextProviderProps = {
   // none
 };
 
+const useIsConnected = (stanWasmServerUrl: string) => {
+  const probeUrl = `${stanWasmServerUrl}/probe`;
+  const [isConnected, setIsConnected] = useState<boolean>(false);
+  const [retryCode, setRetryCode] = useState<number>(0);
+  const retryConnection = useCallback(() => {
+    setRetryCode((r) => r + 1);
+  }, []);
+  useEffect(() => {
+    setIsConnected(false);
+    if (!probeUrl.startsWith("http://") && !probeUrl.startsWith("https://")) {
+      // important to do this check because otherwise fetch may succeed because
+      // the server of this web app may respond with success
+      return;
+    }
+    (async () => {
+      try {
+        const response = await fetch(probeUrl);
+        if (response.status === 200) {
+          setIsConnected(true);
+        }
+      } catch (err) {
+        setIsConnected(false);
+      }
+    })();
+  }, [probeUrl, retryCode]);
+  return { isConnected, retryConnection };
+};
+
```

Comment:
Don't mean to open up a can of worms here by suggesting a rewrite of working code, but I'm wondering--now that this is moved into the CompileContext--whether this still needs to be a hook, or if it'd be more natural/clear to write it as just a couple functions (with the state/`useState` held direct on the CompileContextProvider, perhaps).

Reply:
I kinda liked the encapsulation, but it does seem a bit odd to have a custom hook which is only used in one place, with that one place being it's own hook-like-thing.

Happy to go either way

Reply:
I don't have strong feelings about it, and it's working as is, so it's probably not worth any mental energy to re-envision it.

You can just ship it :smile_cat: